client/containers/workflow-list/getters.spec.js

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2021 Uber Technologies Inc.
+//
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import { ROUTE_QUERY } from '../route/getter-types';
+import { FILTER_MODE_ADVANCED, FILTER_MODE_BASIC } from './constants';
+import wfListGetterFns from './getters';
+import { WORKFLOW_LIST_FILTER_MODE } from './getter-types';
+import { initGetters } from '~test';
+
+describe('workflow list getters', () => {
+  describe('when calling getters[WORKFLOW_LIST_FILTER_MODE]', () => {
+    describe('and getters[ROUTE_QUERY].filterMode is set', () => {
+      const getterFns = {
+        ...wfListGetterFns,
+        [ROUTE_QUERY]: () => ({
+          filterMode: FILTER_MODE_ADVANCED,
+        }),
+      };
+
+      it('should return the value set.', () => {
+        const getters = initGetters({ getterFns });
+        const output = getters[WORKFLOW_LIST_FILTER_MODE];
+
+        expect(output).toEqual(FILTER_MODE_ADVANCED);
+      });
+    });
+
+    describe('and getters[ROUTE_QUERY].filterMode is not set', () => {
+      const getterFns = {
+        ...wfListGetterFns,
+        [ROUTE_QUERY]: () => ({
+          filterMode: null,
```

Comment:
do you need to set this to FILTER_MODE_BASIC?

Reply:
Good question, you could do either:

Basically the query param `filterMode` in the URL is what determines whether `filterMode` is `basic` or `advanced`. If the query param is set to `filterMode=advanced` then it will return `advanced` (or `FILTER_MODE_ADVANCED`). Similarly if its set to `filterMode=basic` it will return `basic` (or `FILTER_MODE_BASIC`). When this query param is not set, i.e. `filterMode` is not within the URL at all, it will default to `basic`. Since this is what we are trying to test in this test that if no `filterMode` is set in the URL it will default to `basic`. 

See LN28 in [client/containers/workflow-list/getters.js](https://github.com/uber/cadence-web/pull/303/files/217be9ccd01445be34cfeb841340da9b882f6bfa#diff-c2524d9ec127de26428523487f93406279c410da9b3aa9f1ea9e55bf46809c37)
```
getters[ROUTE_QUERY].filterMode || FILTER_MODE_BASIC,
```